### PR TITLE
Python - Attic&Button -  Do not merge

### PR DESCRIPTION
**Pull request by oakbani**

### CODE REVIEW

File: public/js/template-routes.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+$( "#loginForm" ).submit(function( event ) {
+    event.preventDefault();
+    var param = "userId="+$("#exampleInputEmail1").val();
+    if(loginUserRequest(param)){
+        currentUser = {"userName": getCurrentUserRequest()};
+        $('#profileContent').html(loadTemplate( layoutTemplates['profile'], currentUser));
+        $('#loginModal').modal('hide');
+    }else{
+        alert("Invalida email!");
```
Comment (oakbani) [inline public/js/template-routes.js]: @rashidsp typo

File: public/templates/checkout-content.hbs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      <div class="card">
+         <div class="card-header bg-dark text-white">
+            SHIPPING INFO
+         </div>
+         <div class="card-body">
+            <form>
+               <div class="form-row">
+                  <div class="form-group col-md-6">
+                     <label for="inputEmail4">First Name</label>
+                     <input class="form-control" id="inputFirstName" placeholder="First Name" type="text" value="Rashidsp204">
```
Comment (oakbani) [inline public/templates/checkout-content.hbs]: Please fix value

File: public/index.html [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+             <div class="modal-header card-header">
+                <h5 class="modal-title" id="receiptModalLongTitle">Receipt # xxx</h5>
+                <button aria-label="Close" class="close" data-dismiss="modal" id="closeCartModal" type="button">
+                <span aria-hidden="true">x</span>
+                </button>
+             </div>
+             <div class="modal-body card-block" id="receiptTemplate">
+             </div>
+             <div class="modal-footer">
+                <button type='buttton' class="btn btn-info btn-lg checkout" onclick="buyNowClear();">OK</button>
```
Comment (oakbani) [inline public/index.html]: Replace OK, by Close

File: public/templates/home-content.hbs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+                        <ul class="list-group list-group-flush">
+                            <li class="list-group-item">Color: {{this.color}}</li>
+                            <li class="list-group-item">Category: {{this.category}}</li>
+                        </ul>
+                        <p></p>
+                        <div class="row">
+                            <div class="col-xs-12 col-md-7">
+                                <p class="lead">
+                                    {{#ifGreater discountPercentage 0 }}
+                                        <del class="text-danger pr-2">${{discount}}</del>
```
[the remark this answers is not in the document]
Comment (rashidsp) [reply public/templates/home-content.hbs]: Done

File: public/templates/home-content.hbs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+            <div class="item col-xs-3 col-lg-3" style="margin-bottom:45px;">
+                <div class="thumbnail">
+                    <img class="img-fluid group list-group-image" style="width: 100%;height: 240px;" src={{this.image_url}} alt="Item 2">
+                    <div class="caption">
+                        <h5 class="group inner list-group-item-heading">
+                             {{this.name}}
+                        </h5>
+                        <p class="group inner list-group-item-text"></p>
+                        <ul class="list-group list-group-flush">
+                            <li class="list-group-item">Color: {{this.color}}</li>
```
[the remark this answers is not in the document]
Comment (rashidsp) [reply public/templates/home-content.hbs]: Done

File: public/templates/checkout-content.hbs (hunk 2 above)
Comment (rashidsp) [reply public/templates/checkout-content.hbs]: Done

File: public/templates/checkout-content.hbs [hunk 6]
```diff
@@ -0,0 +1,61 @@
+<div class="card-header text-center">
+   Variation: {{variation}}
```
[the remark this answers is not in the document]
Comment (rashidsp) [reply public/templates/checkout-content.hbs]: Done

File: public/templates/cart-content.hbs [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+		</tr>
+		<tr>
+			<td class="text-right border-0"></td>
+			<th class="border-0 text-right" colspan="4">Grand Total</th>
+			<td class="border-0 text-right totals-value" id="cartTotal">{{total}}</td>
+		</tr>
+		<tr>
+			<td class="border-0 text-left" colspan="4">
+				<input id="clearCart" class="btn btn-lg btn-danger" type="button" value="Clear" onclick="clearCart();"/>
+				<a data-confirm="Are you sure? want to delete all items!"
```
[the remark this answers is not in the document]
Comment (rashidsp) [reply public/templates/cart-content.hbs]: Done

File: public/templates/cart-content.hbs [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+						<span>Color: {{this.product.color}},</span>
+						<span>Category: {{this.product.category}}</span>
+						</p>
+					</td>
+					<td class="text-right product-quantity" title="Amount">
+						<input max="100" min="1" actual_qty= "{{this.quantity}}" id="pr-qty_{{this.product.id}}" style="width: 2.5rem;" type="number" value="{{this.quantity}}", onchange="updateCart({{this.product.id}});">
+					</td>
+					<td class="text-right product-price totals-value" title="Price">{{this.product.price}}</td>
+					<td class="text-right product-line-price totals-value" title="Total">{{this.total}}</td>
+					<td class="product-removal text-right" title="delete">
```
[the remark this answers is not in the document]
Comment (rashidsp) [reply public/templates/cart-content.hbs]: Done

File: public/templates/cart-content.hbs [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+							{{this.product.name}}
+						</strong>
+						</p>
+						<p>
+						<span>Color: {{this.product.color}},</span>
+						<span>Category: {{this.product.category}}</span>
+						</p>
+					</td>
+					<td class="text-right product-quantity" title="Amount">
+						<input max="100" min="1" actual_qty= "{{this.quantity}}" id="pr-qty_{{this.product.id}}" style="width: 2.5rem;" type="number" value="{{this.quantity}}", onchange="updateCart({{this.product.id}});">
```
[the remark this answers is not in the document]
Comment (rashidsp) [reply public/templates/cart-content.hbs]: Done

File: public/js/template-routes.js [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+        $('#profileContent').html(loadTemplate(layoutTemplates['profile'],null));
+    }else{
+        alert("Failed!");
+    }
+};
+
+function addToCart(product_id, qty, fromShopPage) {
+    var params = {"productId":product_id, "quantity": qty, "fromShopPage": fromShopPage}
+    jsonResponse = addToCartRequest(params);
+    var message = {"message": (jsonResponse ? "Product added to cart!" : "Failed to add product into cart!")};
```
[the remark this answers is not in the document]
Comment (rashidsp) [reply public/js/template-routes.js]: Done

File: public/index.html (hunk 3 above)
Comment (rashidsp) [reply public/index.html]: Done

File: public/js/template-routes.js (hunk 1 above)
Comment (rashidsp) [reply public/js/template-routes.js]: Done

**Comment by rashidsp** [comment]
@oakbani Please review the changes.

**Comment by oakbani** [comment]
Thanks. I will review tomorrow

**Comment by oakbani** [comment]
@rashidsp  Merging this. Please create another branch from demo-app-review, do kelly's changes andmake similar PR

**Comment by oakbani** [comment]
@msohailhussain  FYI

**Pull request merged**